### PR TITLE
libunwind: 1.6.2 -> 1.7.2

### PR DESCRIPTION
**Pull request by jiegec**
## Description of changes

The original upstream no longer publishs new tarballs, switch to GitHub.

Remove upstreamed patch.

Add new platforms supported by latest libunwind.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libunwind/default.nix [hunk 1]
```diff
@@ -2,21 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "libunwind";
-  version = "1.6.2";
+  version = "1.7.2";
 
   src = fetchurl {
-    url = "mirror://savannah/libunwind/${pname}-${version}.tar.gz";
-    sha256 = "sha256-SmrsZmmR+0XQiJxErt6K1usQgHHDVU/N/2cfnJR5SXY=";
+    url = "https://github.com/libunwind/libunwind/releases/download/v${version}/libunwind-${version}.tar.gz";
```
Comment (pbsds) [inline pkgs/development/libraries/libunwind/default.nix:8]: Should we use their provided tarballs, or should we rather use `fetchFromGitHub`? I can't seem to find any CI job creating those tarballs.

**Review by pbsds — CHANGES_REQUESTED** [review]
nixpkgs-review is being a bit grumpy, so a review of the diff will have to do for now:

The switch from gnu savannah to github is mentioned [here](https://github.com/libunwind/libunwind/discussions/629), but not discussed.

* The removed patch is tagged in 1.7.2
* loongarch64-linux is added in https://github.com/libunwind/libunwind/pull/316

changelog:
* https://github.com/libunwind/libunwind/releases/tag/v1.7.0
* https://github.com/libunwind/libunwind/releases/tag/v1.7.1
* https://github.com/libunwind/libunwind/releases/tag/v1.7.2

File: pkgs/development/libraries/libunwind/default.nix (hunk 1 above)
Comment (jiegec) [reply pkgs/development/libraries/libunwind/default.nix]: Updated

**Comment by pbsds** [comment]
LGTM, built 5 reverse dependencies locally and they seem to work.

**Pull request merged**